spec/runner_spec.rb

```diff
@@ -0,0 +1,185 @@
+
+    it 'does not exit' do
+      is_expected.to_not raise_error
+    end
+  end
+
+  context 'when any environment variables are not set' do
+    before(:each) { envvars.each { |var| unset_environment_variable(var) } }
+
+    it 'exits' do
+      is_expected.to raise_error(SystemExit)
+    end
+  end
+
+  after(:each) { restore_env }
+end
+
+describe 'Jacaranda#run' do
+  let(:names) { Array.new(3) { Faker::Name.first_name } }
+  let(:runners) do
+    sorted = names.sort_by { |c| c.to_s.split('::').last }
+    sorted.map { |name| Object.const_set(name, Class.new(Jacaranda::Runner)) }
+  end
+
+  it 'executes all runners by default' do
+    expect(Jacaranda.runners.size).to be >= 2
+  end
+
+  it 'filters to a single runner' do
+    args = %w[--runners] << runners.first.to_s
+    Jacaranda.parse(args)
+    expect(Jacaranda.runners.size).to be 1
+    expect(Jacaranda.runners).to eq([runners.first])
+  end
+
+  it 'filters to multiple runners' do
+    args = %w[--runners] << runners[0..1].join(',')
+    Jacaranda.parse(args)
+    expect(Jacaranda.runners.size).to be 2
+    expect(Jacaranda.runners).to eq(runners[0..1])
+  end
+end
+
+describe '#posted_in_last_fortnight?' do
+  let(:text) { Faker::Lorem.paragraph(2) }
+
+  context 'if record exists' do
+    it do
+      # Fake a successful post
+      Jacaranda::Runner.record_successful_post(text)
+      # Then test
+      expect(Jacaranda::Runner.posted_in_last_fortnight?).to be true
+    end
+  end
+
+  context 'if record does not exist' do
+    it do
+      expect(Jacaranda::Runner.posted_in_last_fortnight?).to be false
+    end
+  end
+
+  context 'if posted > 14 days ago' do
+    it nil, :aggregate_failures do
+      10.times do
+        # Fake a successful post
+        text = Faker::RickAndMorty.quote
+        Jacaranda::Runner.record_successful_post(text)
+        # Test now
+        expect(Jacaranda::Runner.posted_in_last_fortnight?).to be true
+        # Test the future
+        time_travel_to(Date.today + 15.days)
+        expect(Jacaranda::Runner.posted_in_last_fortnight?).to be false
+      end
+    end
+  end
+
+  context 'when there is no schema or data' do
+    it do
+      expect(Jacaranda::Runner.posted_in_last_fortnight?).to be false
+    end
+  end
+end
+
+describe '#posts' do
+  let(:names) { Array.new(3) { Faker::Name.first_name } }
+  let(:runners) do
+    sorted = names.sort_by { |c| c.to_s.split('::').last }
+    sorted.map { |name| Object.const_set(name, Class.new(Jacaranda::Runner)) }
+  end
+  let(:count) { 10 }
+
+  it 'only returns posts for the runner type', :aggregate_failures do
+    runners.each do |runner|
+      (1..count).each do |n|
+        text = Faker::RickAndMorty.quote
+        time_travel_to(n.days.ago) { runner.record_successful_post(text) }
+      end
+    end
+
+    runners.each do |runner|
+      expect(runner.posts.size).to eq(count)
+    end
+  end
+
+  after(:each) { ScraperWiki.sqliteexecute('DELETE FROM data') }
+end
+
+describe '#run' do
```

Comment:
This should be nested under the Class it's part of: 

describe "Jacaranda" do
  describe ".posts" do
    ...
  end

  describe ".run" do
    ...
  end
end